core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -426,48 +351,18 @@ export class ContributorStatsComponent {
     };
   }
 
-  sliceStatsIntoPage(
-      statsForPage: TranslationContributionStats[] |
-        TranslationReviewStats[] |
-        QuestionContributionStats[] |
-        QuestionReviewStats[]): Stats[] {
-    let slicedStats: Stats[] = [];
-    const iterations: number = Math.floor(
-      statsForPage.length / this.ITEMS_PER_PAGE);
-
-    for (let i = 0; i <= iterations; i++) {
-      slicedStats.push({
-        pageNumber: i + 1,
-        stats: statsForPage.slice(
-          (i * this.ITEMS_PER_PAGE),
-          (i * this.ITEMS_PER_PAGE) + this.ITEMS_PER_PAGE)
-      });
-    }
-
-    return slicedStats;
-  }
-
-  generatePagedStats(slicedStats: Stats[], language?: string): PagedStats {
-    let pagedStats: PagedStats = {
-      currentPage: 1,
-      totalPages: slicedStats.length,
-      pagedStats: slicedStats
-    };
-
-    if (language) {
-      pagedStats.language =
-        this.languageUtilService.getAudioLanguageDescription(language);
-    }
-
-    return pagedStats;
+  nextPage(page: Stats): void {
+    page.currentStartingIndex += this.ITEMS_PER_PAGE;
+    page.currentEndingIndex += this.ITEMS_PER_PAGE;
   }
 
-  nextPage(block: PagedStats): void {
-    block.currentPage += 1;
+  previousPage(page: Stats): void {
+    page.currentStartingIndex -= this.ITEMS_PER_PAGE;
+    page.currentEndingIndex -= this.ITEMS_PER_PAGE;
   }
 
-  previousPage(block: PagedStats): void {
-    block.currentPage -= 1;
+  columnSortDirection(): number {
```

Comment:
All method names should start with verbs. Ditto above (previousPage, nextPage, etc.) -- please fix.

Reply:
Done